shell/platform/android/io/flutter/embedding/engine/mutatorsstack/FlutterMutatorView.java

```diff
@@ -0,0 +1,89 @@
+package io.flutter.embedding.engine.mutatorsstack;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Matrix;
+import android.graphics.Path;
+import android.widget.FrameLayout;
+import androidx.annotation.NonNull;
+
+public class FlutterMutatorView extends FrameLayout {
+
+  private FlutterMutatorsStack mutatorsStack;
+  private float screenDensity;
+  private int left;
+  private int top;
+
+  public FlutterMutatorView(@NonNull Context context) {
+    super(context, null);
+  }
+
+  public void readyToDisplay(
+      @NonNull FlutterMutatorsStack mutatorsStack,
+      float screenDensity,
+      int left,
+      int top,
+      int width,
+      int height) {
+    this.mutatorsStack = mutatorsStack;
+    this.screenDensity = screenDensity;
+    this.left = left;
+    this.top = top;
+    FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(width, height);
+    layoutParams.leftMargin = left;
+    layoutParams.topMargin = top;
+    setLayoutParams(layoutParams);
+    setClipChildren(false);
+    setWillNotDraw(false);
+  }
+
+  @Override
+  public void draw(Canvas canvas) {
```

Comment:
From https://developer.android.com/reference/android/view/View#draw(android.graphics.Canvas)

> When implementing a view, implement onDraw(android.graphics.Canvas) instead of overriding this method

Reply:
We to have override draw here as we need to clip the current view's canvas. Doing the same thing at `onDraw` doesn't seem to work.